index.js

```diff
@@ -0,0 +1,18 @@
+module.exports = (api, options) => {
+  options.css.extract = false
```

Comment:
I have been wondering if this is a sensible default. Could this instead set the public path to a default, but allow for setting it through the plugin? I think a consumer could override it anyway, but that'd be undoing what this plugin undid.

Reply:
No -- my understanding is that extracted css must be loaded with a `<link>` element in your HTML file and will not be automatically downloaded even if the public path is correct. Normally, Vue depends on HtmlWebpackPlugin to inject that `<link>` element into the index.html file. In our case, we can't depend on that.

If I'm missing something about Vue's CSS implementation (which I think uses https://webpack.js.org/plugins/mini-css-extract-plugin/), I'd love to learn more 😄

Reply:
Is that the mini-css-extract plugin though? I am no webpack expert. I thought mini-css-extract plugin just creates css files for "On-Demand-Loading of CSS", and its a different plugin that detects the additional chunks and puts them in the html (probably HtmlWebpackPlugin). The "On-Demand-Loading of CSS" part sounds like it would be done with webpack dynamic imports, which might be further backed up by its [usage of the `publicPath` config](https://webpack.js.org/plugins/mini-css-extract-plugin/#publicpath). 

Also I'm extending this understanding to the splitChunks because it should create "on-demand chunks" that can be dynamically loaded... maybe? 😱 I'm only about 50% confident in this now haha 

I guess, what I'm thinking is that it'd be better to disable whatever HtmlWebpackPlugin config, not css.extract or optimization.splitChunks.

Reply:
> Is that the mini-css-extract plugin though?
Yes, it is.

Disabling htmlwebpackplugin breaks the entire build in a way that would be awful to try to fix. I already tried. HtmlWebpackPlugin is deeply deeply embedded as an assumption of vue-cli. Not worth it trying to remove it.

Additionally, disabling htmlwebpackplugin wouldn't actually help. It *does not* make extracted css or optimize.splitChunks work. My understanding is that mini-css-extract-plugin *does not create a dynamic import* to load the css. Instead, it relies on you loading the css manually in an html file.

I verified this by making the css extracted and then loading the vue app into single-spa-playground. Notice how the screenshot below has all the styles missing. This is due to the css file not being loaded. In my example, I am not using HtmlWebpackPlugin.

<img width="506" alt="image" src="https://user-images.githubusercontent.com/5524384/61750470-bcf3b800-ad62-11e9-9655-a9fb626a593d.png">

Also, I created a small webpack test repo that shows that when you use mini-css-extract-plugin that *the js bundle does not attempt to download it at all*. See following screenshots:

<img width="745" alt="image" src="https://user-images.githubusercontent.com/5524384/61750889-d5b09d80-ad63-11e9-89ea-f4069826c595.png">

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/5524384/61750899-dc3f1500-ad63-11e9-8be9-3a86ab3cdcd1.png">

<img width="403" alt="image" src="https://user-images.githubusercontent.com/5524384/61750921-eeb94e80-ad63-11e9-9d86-5dc6281ed0b7.png">

All of this illustrates that this has nothing to do with public path. It has everything to do with how extracting the css means that the css won't be loaded at all unless injected into the html file. Which is not a good option for single-spa.

Reply:
Ok, I won't block this then. I think this changes/breaks users expectations (the CSS is included in their JS, no chunk optimization). We should document that. Or... How difficult would it be to make dynamic imports work out of the box with this plugin?

Reply:
Sorry my ignorance. Looks like dynamic imports is as easy as `import('theThing.js')`. I'm failing to uderstand how CRA does it though, since the syntax there is `import 'theThing.css'`. Also, how would that play into the SFC setup...

Reply:
> I think this changes/breaks users expectations (the CSS is included in their JS, no chunk optimization)

I disagree pretty strongly with this. CSS included in their JS is the default for all Vue CLI users when they do `vue build --target lib`. A single-spa application is a lib, and it's expected for the css to be embedded within it. Additionally, no chunk optimizations does not mean that we aren't doing code splits. Also it doesn't mean that there isn't optimization happening.

As for dynamic imports, those require setting the public path but then nothing else. People who use Vue shoud definitely be using dynamic import.

Reply:
If you have an alternative proposal, I'd be happy to review a PR. I think the discussion here has been beneficial, but you don't actually have an alternative proposal. This approach is the standard one for both the single-spa-react and single-spa-angular ecosystems, so I'm confident it will work well for Vue users, too.